### PR TITLE
Chore: Solidity coverage

### PR DESCRIPTION
**Pull request by alexroan**
* New script in `contracts/` to run coverage check: `yarn coverage`.
* CI runs this new script and ensures it doesn't fail. However, it does nothing with the coverage results.

### CODE REVIEW

File: contracts/.solcover.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  mocha: {
+    grep: "@skip-coverage", // Find everything with this tag
```
Comment (alexroan) [inline contracts/.solcover.js:3]: Given `solidity-coverage` compiles the contract without the optimizer, we use this tag on any tests that rely on gas measurements.

File: .github/workflows/solidity.yml [hunk 2]
```diff
@@ -29,6 +29,34 @@ jobs:
...
+            ~/.cache
+            **/node_modules
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - run: yarn install --frozen-lockfile
+      - name: Run coverage
+        run: ./tools/ci/solidity_coverage
```
Comment (alexroan) [inline .github/workflows/solidity.yml:58]: The coverage script is run here to make sure no errors occur, but we don't do anything with the percentage outputs at the moment.

**Comment by archseer** [comment]
You might want to exclude `v*/tests/ ` from the coverage report so it doesn't skew the totals?

**Comment by archseer** [comment]
Could this be hooked up through codecov? That way we don't necessarily have to turn it into a required CI check, but there would still be useful stats available on PRs (for example a big drop in coverage)

**Comment by alexroan** [comment]
> You might want to exclude v*/tests/ from the coverage report so it doesn't skew the totals?

@archseer Unfortunately, excluding these skews the totals even more. Many of the code paths in the tests go through mocks and test helper contracts, so excluding them means that solidity-coverage ignores those paths altogether 😕

**Review by archseer — APPROVED** [review]

**Pull request merged**